geometry_msgs/Pose currentPose
nav_msgs/OccupancyGrid ProjectedGrid
---
geometry_msgs/Pose nextPose